src/frontend/js/types/Joanie.ts

```diff
@@ -138,8 +138,8 @@ export interface CourseProduct extends Product {
 
 export interface DefinitionResourcesProduct {
   id: Product['id'];
-  certificate_definition_id: CertificateDefinition['id'];
-  contract_definition_id: ContractDefinition['id'];
+  certificate_definition_id?: CertificateDefinition['id'];
+  contract_definition_id?: ContractDefinition['id'];
```

Comment:
These are not optional but nullable.

```suggestion
  certificate_definition_id: Nullable<CertificateDefinition['id']>;
  contract_definition_id: Nullable<ContractDefinition['id']>;
```